packages/firebase_auth/firebase_auth/example/android/app/build.gradle

```diff
@@ -1,60 +1,72 @@
 def localProperties = new Properties()
 def localPropertiesFile = rootProject.file('local.properties')
 if (localPropertiesFile.exists()) {
-  localPropertiesFile.withReader('UTF-8') { reader ->
-    localProperties.load(reader)
-  }
+    localPropertiesFile.withReader('UTF-8') { reader ->
+        localProperties.load(reader)
+    }
 }
 
 def flutterRoot = localProperties.getProperty('flutter.sdk')
 if (flutterRoot == null) {
-  throw new GradleException('Flutter SDK not found. Define location with flutter.sdk in the local.properties file.')
+    throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
 }
 
 def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
 if (flutterVersionCode == null) {
-  flutterVersionCode = '1'
+    flutterVersionCode = '1'
 }
 
 def flutterVersionName = localProperties.getProperty('flutter.versionName')
 if (flutterVersionName == null) {
-  flutterVersionName = '1.0'
+    flutterVersionName = '1.0'
 }
 
 apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-  compileSdkVersion 29
-
-  lintOptions {
-    disable 'InvalidPackage'
-  }
-
-  defaultConfig {
-    applicationId 'io.flutter.plugins.firebaseauthexample'
-    minSdkVersion 21
-    targetSdkVersion 28
-    versionCode flutterVersionCode.toInteger()
-    versionName flutterVersionName
-    testInstrumentationRunner 'androidx.test.runner.AndroidJUnitRunner'
-  }
-
-  buildTypes {
-    release {
-      // TODO: Add your own signing config for the release build.
-      // Signing with the debug keys for now, so `flutter run --release` works.
-      signingConfig signingConfigs.debug
+    compileSdkVersion flutter.compileSdkVersion
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
```

Comment:
After updating Gradle plugin to 7.0.4 as suggested a bit lower there will be no need in specifying Java 8, since in 7.0.4 plugin Java 11 is required.